src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/HistogramAdvice.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace System.Diagnostics.Metrics
+{
+    /// <summary>
+    /// Contains settings used to advise metrics consumers how to construct storage for <see cref="Histogram{T}"/> instruments.
+    /// </summary>
+    /// <typeparam name="T">Histogram value type.</typeparam>
+    public sealed class HistogramAdvice<T> where T : struct
+    {
+        /// <summary>
+        /// Constructs a new instance of <see cref="HistogramAdvice{T}"/>.
+        /// </summary>
+        /// <remarks>
+        /// Notes:
+        /// <list type="bullet">
+        /// <item>An empty set of bucket boundaries hints that the histogram by default should NOT contain buckets and should only track count and sum values.</item>
+        /// <item>A set of distinct increasing values for bucket boundaries hints that the histogram should use those for its default bucket configuration.</item>
+        /// </list>
+        /// </remarks>
+        /// <param name="explicitBucketBoundaries">Explicit bucket boundaries advised to be used with the histogram.</param>
+        public HistogramAdvice(IEnumerable<T> explicitBucketBoundaries)
```

Comment:
In the future if we have multiple kinds of advice how do we imagine this scaling up? I'm guessing constructor overloads start getting unwieldy once we have more than 3-4 parameters and we'd switch over to settable properties instead?